models/git_diff.go

```diff
@@ -252,19 +252,27 @@ func ParsePatch(maxLines, maxLineCharacters, maxFiles int, reader io.Reader) (*D
 	input := bufio.NewReader(reader)
 	isEOF := false
 	for !isEOF {
-		line, err := input.ReadString('\n')
-		if err != nil {
-			if err == io.EOF {
-				isEOF = true
-			} else {
-				return nil, fmt.Errorf("ReadString: %v", err)
+		var linebuf bytes.Buffer
+		for {
```

Comment:
From what I can tell, we only break from this loop when we reach an EOF or new-line. Should we break once `linebuf.Len() >= maxLineCharacters`? Otherwise it's not clear to me how this helps memory usage.

Reply:
@ethantkoenig I think it's right. See line 272.

Reply:
Line 272 sets `curFile.IsIncomplete` to true, but the loop will still run for more iterations.

Reply:
Yes, so that, we could find the next line.

Reply:
Yeah, first I used `break` to stop reading, but that would cause https://try.gitea.io/mrexodia/DarkSouls3.TextViewer/commit/629cf9b3d6b295bbcddf76d1f6167259b764d9dc to crash because it would think the next diff line was at the next character.